### ISSUE TITLE
PCC drop in deformable DETR - deformable_detr_single_scale_model

### ISSUE THREAD

**Issue by ctr-lelanchelianTT** [tenstorrent/tt-xla#3951]
## Describe the bug
 
- Deformable DETR model execution produces a final PCC of ~0.52, failing accuracy requirements - full model log [deformable_detr_single_scale_model.log](https://github.com/user-attachments/files/26338958/deformable_detr_single_scale_model.log)
- The failure was isolated to the `torch.nn.functional.grid_sample` operation in the encoder -  log 
[deformable_detr_single_scale_model_grid_sample.log](https://github.com/user-attachments/files/26339040/deformable_detr_single_scale_model_grid_sample.log)
- torch.nn.functional.grid_sample sanity log - [sanity_grid_sample.log](https://github.com/user-attachments/files/26347462/sanity_grid_sample.log), [test_grid_sample_deformable_detr.py](https://github.com/user-attachments/files/26347422/test_grid_sample_deformable_detr.py)

- `grid_sample` is lowered to ~100 TTNN operations in `grid_codegen/main.py`
- The root cause is floating point rounding drift in `ttnn.matmul` used for index flattening, causing incorrect pixel lookups during embedding - log [sanity_ttnn_matmul.log](https://github.com/user-attachments/files/26339155/sanity_ttnn_matmul.log)

**Comment by saiarthiraguram** [comment]
## Repro

```
git checkout sai_arthi_raguram/grid_sample
git submodule update --init --recursive
 
# Full model run
pytest tests/runner/test_models.py::test_all_models_torch[deformable_detr/pytorch-Deformable_Detr-single_device-inference] -svv

# sanity run
pytest tests/torch/ops/test_grid_sample_deformable_detr.py -svv 

# Index flattening op audit
python tests/torch/ops/test_pcc_debug.py -svv
